Algorithm.Framework/Alphas/MacdAlphaModel.cs

```diff
@@ -116,31 +127,17 @@ class SymbolData
             public readonly Security Security;
             public readonly IDataConsolidator Consolidator;
             public readonly MovingAverageConvergenceDivergence MACD;
+            public TimeSpan DataResolution => Security.Resolution.ToTimeSpan();
 
-            public SymbolData(QCAlgorithmFramework algorithm, Security security, TimeSpan period)
+            public SymbolData(QCAlgorithmFramework algorithm, Security security, int fastPeriod, int slowPeriod, int signalPeriod, MovingAverageType movingAverageType)
             {
                 Security = security;
-                Consolidator = algorithm.ResolveConsolidator(security.Symbol, period);
+                Consolidator = algorithm.ResolveConsolidator(security.Symbol, security.Resolution);
                 algorithm.SubscriptionManager.AddConsolidator(security.Symbol, Consolidator);
 
-                MACD = new MovingAverageConvergenceDivergence(12, 26, 9, MovingAverageType.Exponential);
-
-                Consolidator.DataConsolidated += OnDataConsolidated;
-            }
+                MACD = algorithm.MACD(security.Symbol, fastPeriod, slowPeriod, signalPeriod, movingAverageType);
 
-            /// <summary>
-            /// Cleans up the indicator and consolidator
-            /// </summary>
-            /// <param name="algorithm">The algorithm instance</param>
-            public void CleanUp(QCAlgorithmFramework algorithm)
```

Comment:
Removed on decision to reduce requirements of people using consolidators and self-cleaning up.